### PR TITLE
{bio}[foss/2017a] Trinity 2.4.0 (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**
intel/2017a fails with inchworm
similar errors like:
```
GCCcore/6.3.0/bin/../include/c++/6.3.0/bits/stl_multimap.h(972): note: this candidate was rejected because at least one template argument could not be deduced
      operator==(const multimap<_Key, _Tp, _Compare, _Alloc>& __x,
      ^
```
I did not have time to look into more details into the problem, bit it might be intel 17.0 compiler problem/C++ incompatibility
See https://software.intel.com/en-us/forums/intel-c-compiler/topic/703133

### CODE REVIEW

**Comment by hajgato** [comment]
The actual error is:
```
./analysis/sequenceUtil.cc(53): error: no operator "==" matches these operands
            operand types are: std::ifstream == int
    if (fileReader == 0) { // couldn't open file
                   ^
```

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 13 out of 13 (5 easyconfigs in this PR)
ca005.camp.thecrick.org - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2640 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/7fc973d10d46a8f980b1b29cc7ae156b for a full test report.

File: easybuild/easyconfigs/b/Bowtie/Bowtie-1.1.2-GCCcore-6.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+homepage = 'http://bowtie-bio.sourceforge.net/index.shtml'
+description = """Bowtie is an ultrafast, memory-efficient short read aligner.
+It aligns short DNA sequences (reads) to the human genome.
+"""
+
+toolchain = {'name': 'GCCcore', 'version': '6.3.0'}
+toolchainopts = {'pic': True}
+
+sources = ['%(namelower)s-%(version)s-src.zip']
+source_urls = ['http://download.sourceforge.net/bowtie-bio/']
```
Comment (verdurin) [inline easybuild/easyconfigs/b/Bowtie/Bowtie-1.1.2-GCCcore-6.3.0.eb]: @hajgato please add a checksum here

File: easybuild/easyconfigs/b/Bowtie2/Bowtie2-2.3.2-foss-2017a.eb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ to long reference sequences. It is particularly good at aligning reads of about 50 up to 100s or 1,000s 
+ of characters, and particularly good at aligning to relatively long (e.g. mammalian) genomes. 
+ Bowtie 2 indexes the genome with an FM Index to keep its memory footprint small: for the human genome, 
+ its memory footprint is typically around 3.2 GB. Bowtie 2 supports gapped, local, and paired-end alignment modes."""
+
+toolchain = {'name': 'foss', 'version': '2017a'}
+toolchainopts = {'pic': True, 'cstd': 'gnu++98'}
+
+sources = ['%(namelower)s-%(version)s-source.zip']
+source_urls = [('http://sourceforge.net/projects/bowtie-bio/files/%(namelower)s/%(version)s', 'download')]
```
Comment (verdurin) [inline easybuild/easyconfigs/b/Bowtie2/Bowtie2-2.3.2-foss-2017a.eb]: Again, checksum here please.

File: easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+homepage = 'http://trinityrnaseq.github.io'
+description = """Trinity represents a novel method for the efficient and robust de novo reconstruction
+ of transcriptomes from RNA-Seq data. Trinity combines three independent software modules: Inchworm,
+ Chrysalis, and Butterfly, applied sequentially to process large volumes of RNA-Seq reads."""
+toolchain = {'name': 'foss', 'version': '2017a'}
+toolchainopts = {'optarch': True}
+
+source_urls = ['https://github.com/trinityrnaseq/trinityrnaseq/archive/']
+sources = ['%(name)s-v%(version)s.tar.gz']
```
Comment (verdurin) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb]: Checksum, please.

File: easybuild/easyconfigs/t/tbb/tbb-2017_U6-GCCcore-6.3.0.eb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+version = '2017_U6'
+
+homepage = 'https://01.org/tbb/'
+description = """Intel(R) Threading Building Blocks (Intel(R) TBB) lets you easily write parallel C++ programs that
+ take full advantage of multicore performance, that are portable, composable and have future-proof scalability."""
+
+toolchain = {'name': 'GCCcore', 'version': '6.3.0'}
+
+source_urls = ['https://github.com/01org/tbb/archive/']
+sources = ['%(version)s.tar.gz']
```
Comment (verdurin) [inline easybuild/easyconfigs/t/tbb/tbb-2017_U6-GCCcore-6.3.0.eb]: And a checksum here too, please.

**Comment by hajgato** [comment]
@verdurin Is it a new policy that we add checksums everywhere?

**Comment by verdurin** [comment]
@hajgato yes it is - I was caught out initially as well

**Comment by hajgato** [comment]
@verdurin It will be fun to make `Perl`, `Python`, `R`, etc. modules ;)

File: easybuild/easyconfigs/b/Bowtie/Bowtie-1.1.2-GCCcore-6.3.0.eb (hunk 1 above)
Comment (hajgato) [inline easybuild/easyconfigs/b/Bowtie/Bowtie-1.1.2-GCCcore-6.3.0.eb]: added

File: easybuild/easyconfigs/b/Bowtie2/Bowtie2-2.3.2-foss-2017a.eb (hunk 2 above)
Comment (hajgato) [inline easybuild/easyconfigs/b/Bowtie2/Bowtie2-2.3.2-foss-2017a.eb]: added

File: easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb (hunk 3 above)
Comment (hajgato) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb]: added

File: easybuild/easyconfigs/t/tbb/tbb-2017_U6-GCCcore-6.3.0.eb (hunk 4 above)
Comment (hajgato) [inline easybuild/easyconfigs/t/tbb/tbb-2017_U6-GCCcore-6.3.0.eb]: added

File: easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb [hunk 5]
```diff
@@ -0,0 +1,35 @@
+name = 'Trinity'
+version = '2.4.0'
+java = 'Java'
+javaver = '1.8.0_121'
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb]: @hajgato no need for this, you can use `%(javaver)s`:

```python
dependencies = [
    ('Java', '1.8.0_121', '', True),
    ('ant', '1.10.1', '-Java-%(javaver)s', True),

Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb]: @hajgato no need for this, you can use `%(javaver)s`:

```python
dependencies = [
    ('Java', '1.8.0_121', '', True),
    ('ant', '1.10.1', '-Java-%(javaver)s', True),

Comment (hajgato) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb]: @boegel done!

File: easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+patches = [
+    'chrysalis_commandline_noconsts_2012-10-05.patch',
+]
+
+builddependencies = [
+    ('Autotools', '20150215', '', ('GCCcore', '6.3.0')),
+]
+
+dependencies = [
+    (java, javaver, '', True),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb]: @hajgato You missed a spot, this should be:

```python
('Java', '1.8.0_121', '', True),

Comment (hajgato) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.4.0-foss-2017a.eb]: @boegel done (Sorry its too morning...)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2107.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/1c44eb2c3505e48749365af9e3f0bf88 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2421.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/09aabc691120226c44df0945f856fdf5 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Pull request merged**